### PR TITLE
feat(database schema) rename all foreign key constraints

### LINKED ISSUE #4793: Name all foreign key constraints
As we have a lot of bhima instance running, having the same database structure facilitate the migration.
Imagine we want to drop a foreign key  from this table bellow 

```sql
CREATE TABLE `table_name` (
  `uuid` BINARY(16),
  `patient_uuid` BINARY(16),
  FOREIGN KEY (`patient_uuid`) REFERENCES `parent` (`uuid`)
) ENGINE=InnoDB DEFAULT CHARACTER SET = utf8mb4 DEFAULT COLLATE = utf8mb4_unicode_ci;

```
In this case you will have to run 
```sql
  SHOW CREATE TABLE `table_name`
```
In order to know the constraint name, but you can see here how it is easy when the constraint name is defined

```sql
CREATE TABLE `table_name` (
  `uuid` BINARY(16),
  `patient_uuid` BINARY(16),
  CONSTRAINT `table_name__parent` FOREIGN KEY (`patient_uuid`) REFERENCES `parent` (`uuid`)
) ENGINE=InnoDB DEFAULT CHARACTER SET = utf8mb4 DEFAULT COLLATE = utf8mb4_unicode_ci;

```
I can drop the FK by running 
```sql
 ALTER TABLE table_name DROP FOREING KEY table_name__parent

```

### PR DESCRIPTION
**Pull request by jeremielodi**
closes #4793

### CODE REVIEW

File: server/models/migrations/next/migrate.sql [hunk 1]
```diff
@@ -1 +1,643 @@
...
+CLOSE curs1;
+END$$
+DELIMITER ;
+
+
+    SET foreign_key_checks = 0;
+
+    CALL drop_constraints('account');
+    ALTER TABLE `account` ADD CONSTRAINT `account_account_type` FOREIGN KEY (`type_id`) REFERENCES `account_type` (`id`),
+    ADD CONSTRAINT `account_entreprise` FOREIGN KEY (`enterprise_id`) REFERENCES `enterprise` (`id`),
```
Comment (jmcameron) [inline server/models/migrations/next/migrate.sql]: Is 'account_entreprise' spelled correctly?

Comment (jeremielodi) [reply server/models/migrations/next/migrate.sql]: It was a typo

File: server/models/migrations/next/migrate.sql [hunk 2]
```diff
@@ -1 +1,644 @@
...
+    ADD CONSTRAINT `medical_sheet__survey_data` FOREIGN KEY (`survey_data_uuid`) REFERENCES `survey_data` (`uuid`),
+    ADD CONSTRAINT `medical_sheet__patient` FOREIGN KEY (`patient_uuid`) REFERENCES `patient` (`uuid`);
+
+    
+    CALL drop_constraints('configuration_analysis_tools');
+    ALTER TABLE `configuration_analysis_tools` 
+    ADD CONSTRAINT `config_analysis_tools__acc_ref` FOREIGN KEY (`account_reference_id`) REFERENCES `account_reference` (`id`),
+    ADD CONSTRAINT `config_analysis_tools__analysis_tool_type` FOREIGN KEY (`analysis_tool_type_id`) REFERENCES `analysis_tool_type` (`id`);
+
+    SET foreign_key_checks = 1;
```
Comment (jmcameron) [inline server/models/migrations/next/migrate.sql:657]: Should SQL files end in a new line?   I could not find an authoritative answer to this, but github is marking it with a red symbol.  Not sure github really knows the answer, but github is suspicious.

**Comment by jmcameron** [comment]
How can I test this?

**Comment by jeremielodi** [comment]
The first test is  ``` ./sh/build-database.sh ``` it will just create the database but this time all the  foreign key constraints  will have the name defined in schema.sql it was not the case before.

For the second test, you should have an exist bhima database, the KARAWA one for example, witch is up to date. run the migration script.

to be sure constraints have been renamed run ``` SHOW CREATE TABLE `table_name`   ```

**Review by jmcameron — CHANGES_REQUESTED** [review]
Using Bhima_test, I found several constraints that did not have double underscores.  I'll wait for you to fix that before testing it on the IMCK database.

  CONSTRAINT `cash_currency`
  CONSTRAINT `cashbox_project`
  CONSTRAINT `config_rubric_item_ibfk_1`
  CONSTRAINT `config_rubric_item_ibfk_2`
  CONSTRAINT `discount_account`
  CONSTRAINT `discount_inventory`
  CONSTRAINT `employee_creditor`
  CONSTRAINT `employee_fonction`
  CONSTRAINT `employee_grade`
  CONSTRAINT `employee_patient`
  CONSTRAINT `employee_service`
  CONSTRAINT `rubric_paiement_ibfk_1`
  CONSTRAINT `rubric_paiement_ibfk_2`

**Comment by jeremielodi** [comment]
The goal here is just to  have a name that is easy to understand and witch should be unique for all our databases

**Comment by jmcameron** [comment]
> The goal here is just to have a name that is easy to understand and witch should be unique for all our databases

But you are consistent in using double underscores everywhere else.  I encourage you to do that for these as well, so it is clear and recognizable pattern.

**Comment by jeremielodi** [comment]
okay

**Comment by jniles** [comment]
@jeremielodi what production database are you testing this on?  I keep running into errors on Karawa and Vanga.

**Comment by jeremielodi** [comment]
okay let me work on it today and I will tell you what.

**Comment by jeremielodi** [comment]
I updated IMCK server, it's running on V1.14.0 of bhima.
@jniles make sure you are running the last migrate file I created ( there is ``` SET foreign_key_checks = 0;``` in side)

**Comment by jniles** [comment]
@jeremielodi I'll give you a review when I've gotten the latest IMCK database to give it a try.

**Comment by jeremielodi** [comment]
Okay, no problem. You can also download one on backups.bhi.ma, just ```mysqldump  imck ...```

**Comment by jmcameron** [comment]
Just tested with Bhima_test and it looks good.  I noticed that you changed all of the double-underscores to single underscores.  Although I thought having double underscores was better because it made it clear which part of the constraint name was the table name.   But I do not object to the way you decided to do it.  I need to test with the IMCK database before finishing my review.

**Comment by jmcameron** [comment]
I have to wait for the IMCK database to be synced up before I can finish the review.

**Review by jmcameron — APPROVED** [review]

**Comment by jmcameron** [comment]
I just double-checked with with the IMCK database it is fine.
@jeremielodi thanks for your work on this (and patience)!

**Comment by jniles** [comment]
It also looks good to me.  Great work @jeremielodi and thanks for the reviews @jmcameron 

bors r=jmcameron,jniles

**Pull request merged**